### ISSUE TITLE
bootstrap error: pathspec 'gnulib' did not match any file(s) known to git

### ISSUE THREAD

**Issue by username30** [jb-alvarado/media-autobuild_suite#1014]
Hello again,

This time I've tried to compile the actually desired build, not just light version and included some packs. I've updated script to last master build and got stuck again at bootstrap step (with another issue though):
![mintty_2019-01-11_04-42-14](https://user-images.githubusercontent.com/24947452/51008089-4e37d580-155b-11e9-9781-c3afd2b8eef5.png)
Here is diagnostic pack: [logs.zip](https://0x0.st/s7OF.zip)

**Comment by 1480c1** [comment]
It seems that you are getting the proper files from the submodules. Try to delete build\gnutls-git and rerun.

**Comment by username30** [comment]
@1480c1 
> Try to delete build\gnutls-git and rerun.
Mmmm... no, I'm still getting exactly the same response:
![mintty_2019-01-11_06-52-45](https://user-images.githubusercontent.com/24947452/51012322-a24bb580-156d-11e9-81cd-42776cf2e01b.png)
Diagnostics: [logs.zip](https://0x0.st/s7VJ.zip)

**Comment by 1480c1** [comment]
Can you see if those particular folders exists? They will be under gnutls-git/tests... if not, can you check if you can see https://gitlab.com/gnutls/gnutls/tree/gnutls_3_6_5/tests/suite/tls-fuzzer and see if you can click on the python-ecdsa and make sure you can accsess it?

**Comment by 1480c1** [comment]
This seems like git could not initialize the submodules, however I was able to successfully do the same just over an hour ago. I can only assume that you may have connection problems to either gitlabs or GitHub, or it could be that your submodules aren't configured properly.

**Comment by username30** [comment]
> Can you see if those particular folders exists? They will be under gnutls-git/tests...

No, there was no "tests" subfolder in "gnutls-git". I created it and copied content of corresponding content from https://gitlab.com/gnutls/gnutls/tree/gnutls_3_6_5/tests and put there content of https://github.com/warner/python-ecdsa , https://github.com/tomato42/tlsfuzzer and https://github.com/tomato42/tlslite-ng
Yet, still getting exactly the same response:

![mintty_2019-01-11_07-26-07](https://user-images.githubusercontent.com/24947452/51013188-2dc74580-1572-11e9-9188-2a26e0d74d4c.png)

[logs.zip](https://0x0.st/s7Vl.zip)

> I can only assume that you may have connection problems to either gitlabs or GitHub, or it could be that your submodules aren't configured properly.

How can I check by myself these points at my side to correctly locate the problem?

**Comment by 1480c1** [comment]
try opening up the mintty shortcut in the base directory and in the window that shows up, type, `cd /trunk/build/gnutls-git` and then try `git fetch` and then `git submodule init`.

**Comment by 1480c1** [comment]
Since you can access those links fine and since you cloned those directories to the folder manually, I can only assume that your submodule configuration is messed up. Usually, the easiest way to fix it is to re-clone the repo, however, you have already deleted and re-cloned gnutls. If possible, can you show the contents of gnutls-git\.git\config?

**Comment by username30** [comment]
@1480c1 

> try opening up the mintty shortcut in the base directory and in the window that shows up, type, cd /trunk/build/gnutls-git and then try git fetch and then git submodule init

Yes, this giving the same error.

> If possible, can you show the contents of gnutls-git.git\config?

Do you mean "gnutls-git\.git\config"? Sure, here it is

![notepad _2019-01-12_01-16-58](https://user-images.githubusercontent.com/24947452/51062375-c65bd500-1607-11e9-84dc-3d1a4a9da990.png)

I believe I can deal somehow this step by hands myself, with no automation, right? I'm just completely new to all this *nix things and barely can understand all these error messages and what they actually stand for. But I assume that script tries to do something, faces some unexpected response with some command and terminating. So... what exactly it tries to do after that
```
gnutls git ............... [Recently updated]
Running bootstrap...
```
lines? What it expect to have as result? How does it check that everything went fine or not? Where is this place in script? How to reproduce it by hands?

**Comment by 1480c1** [comment]
So, I guess one way you can manually fix it is by opening up mintty, then going back to gnutls-git and running git submodule init and see if the config file changes. Here is mine for reference, note the submodules:
```
[core]
	repositoryformatversion = 0
	filemode = false
	bare = false
	logallrefupdates = true
	symlinks = false
	ignorecase = true
[remote "origin"]
	url = https://gitlab.com/gnutls/gnutls.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[submodule "devel/openssl"]
	active = true
	url = https://github.com/openssl/openssl.git
[submodule "gnulib"]
	active = true
	url = https://gitlab.com/libidn/gnulib-mirror.git
[submodule "tests/suite/tls-fuzzer/python-ecdsa"]
	active = true
	url = https://github.com/warner/python-ecdsa
[submodule "tests/suite/tls-fuzzer/tlsfuzzer"]
	active = true
	url = https://github.com/tomato42/tlsfuzzer.git
[submodule "tests/suite/tls-fuzzer/tlslite-ng"]
	active = true
	url = https://github.com/tomato42/tlslite-ng.git
```
![image](https://user-images.githubusercontent.com/8345542/51063929-c5ed1b00-15c2-11e9-84fd-4e0daf89e0c8.png)

**Comment by username30** [comment]
> opening up mintty, then going back to gnutls-git and running git submodule init and see if the config file changes

Mmmm... No, after this command my config file stayed the same as I showed before. I've copied your config file into mine one, but still getting exactly the sage error.

**Comment by 1480c1** [comment]
Try git submodule sync in the same window.

**Comment by username30** [comment]
Still the same. [logs.zip](https://0x0.st/s7xc.zip)

**Comment by 1480c1** [comment]
I was going to recommend that, but I wasn't sure if you wanted gnutls for something such as curl or ffmpeg. But as you already did it, if it works, then it works.

My guess is that since you have already tried manually adding in the submodules but the bootstrap script still doesn't find them, I can only think that it is maybe either a localization issue ([Huge X to doubt](https://i.kym-cdn.com/entries/icons/mobile/000/023/021/e02e5ffb5f980cd8262cf7f0ae00a4a9_press-x-to-doubt-memes-memesuper-la-noire-doubt-meme_419-238.jpg)) or a connection issue with your git not finding the submodules or pulling them (Another huge X). I can only conclude that this is maybe a "Only happens to you" edge case.

**Comment by 1480c1** [comment]
I would say it's okay to close this issue as if someone still has the same problem, either they can comment on this again to "reopen" it or do the same bandaid fix as you did.

**Comment by username30** [comment]
Ok, got it. Anyway, thank you for help! Will go look where will it stuck next time :)

**Comment by 1480c1** [comment]
Good luck

**Comment by Keridos** [comment]
I am running into the same issue here.

**Comment by schmidthubert** [comment]
https://github.com/jb-alvarado/media-autobuild_suite/pull/1102

**Comment by schmidthubert** [comment]
This should be closed then, sorry I forgot to add an auto close keyword!

**Comment by wiiaboo** [comment]
Closed by #1102 .
